src/Fantomas/CodePrinter.fs

```diff
@@ -681,48 +681,128 @@ and genMemberBinding astContext b =
+                        if ctx.Config.AlternativeLongMemberDefinitions then
+                            (indent +> sepNln +> col sepNln ps (genPatWithIdent astCtx) +> unindent) ctx
+                        else
+                            (atCurrentColumn (col sepNln ps (genPatWithIdent astCtx))) ctx
+
+                    expressionFitsOnRestOfLine shortExpr longExpr
+
+                name, parameters
+            | PatTyped(p,t) ->
+                    let n, p = genNameAndParameters p
+                    n +> sepColon +> genType astContext false t, p
+            | PatParen(p) -> genNameAndParameters p
+            | _ -> sepNone, sepNone
+
+        let hasParenthesis = match p with | PatParen _ -> true | _ -> false
+
+        let isSingleTuple =
+            match p with
+            | PatLongIdent(_,_, [_, PatParen(PatTuple(_))], _)
+            | PatLongIdent(_,_, [_, PatParen(PatNamed(_))], _) -> true
+            | _ -> false
+
+        let genName, genParameters = genNameAndParameters p
+
         match e with
         | TypedExpr(Typed, e, t) ->
-            let genName, genParameters, spaceBeforeColon =
-                match p with
-                | PatLongIdent(ao, s, ps, tpso) ->
-                    let aoc = opt sepSpace ao genAccess
-                    let tpsoc = opt sepNone tpso (fun (ValTyparDecls(tds, _, tcs)) -> genTypeParamPostfix astContext tds tcs)
-                    let s = if s = "``new``" then "new" else s
-                    let hasBracket = ps |> Seq.map snd |> Seq.exists hasParenInPat
-                    let multipleParameters = List.length ps > 1
-                    let spaceAfter ctx = onlyIf (ctx.Config.SpaceBeforeMember || multipleParameters || not hasBracket) sepSpace ctx
-                    let name = (aoc -- s +> tpsoc +> spaceAfter)
-
-                    let parameters =
-                        expressionFitsOnRestOfLine
-                            (atCurrentColumn (col sepSpace ps (genPatWithIdent astContext)))
-                            (atCurrentColumn (col sepNln ps (genPatWithIdent astContext)))
-
-                    name, parameters, onlyIf (hasBracket && not multipleParameters) !- " "
-                | _ -> sepNone, sepNone, sepNone
-
             let memberDefinition =
+                let shortExpr =
+                    genName
+                    +> genParameters
+                    +> sepColon
+                    +> genType astContext false t
+                    +> sepEq
+
+                let longExpr ctx =
+                    if ctx.Config.AlternativeLongMemberDefinitions then
+                        (genName
+                         +> genParameters
+                         +> indent
+                         +> sepNln
+                         +> sepColon
+                         +> genType astContext false t
+                         +> sepNln
+                         +> sepEqFixed
+                         +> unindent) ctx
```

Comment:
Can avoid these parenthesis on each branch here, both return a function.

Reply:
I'm not really a fan of using the `<| ctx`, I'm guessing you meant:
```fsharp
                let longExpr ctx =
                    if ctx.Config.AlternativeLongMemberDefinitions then
                        genName
                         +> genParameters
                         +> indent
                         +> sepNln
                         +> sepColon
                         +> genType astContext false t
                         +> sepNln
                         +> sepEqFixed
                         +> unindent
                    else
                        genName +>
                         atCurrentColumn (genParameters
                                          +> sepNln
                                          +> onlyIf isSingleTuple (!- " ")
                                          +> sepColon
                                          +> genType astContext false t
                                          +> sepEq)
                    <| ctx
```
here?

Reply:
Me neither. I usually use a let binding, then apply that.

Reply:
```fsharp
                let longExpr ctx =
                    let expr =
                        if ctx.Config.AlternativeLongMemberDefinitions then
                            genName
                            +> genParameters
                            +> indent
                            +> sepNln
                            +> sepColon
                            +> genType astContext false t
                            +> sepNln
                            +> sepEqFixed
                            +> unindent
                        else
                            genName +>
                            atCurrentColumn (genParameters
                                             +> sepNln
                                             +> onlyIf isSingleTuple (!- " ")
                                             +> sepColon
                                             +> genType astContext false t
                                             +> sepEq)
                    expr ctx
```

Reply:
That looks very clean. :)